packages/web-components/src/tablist/tablist.ts

```diff
@@ -0,0 +1,438 @@
+      return this.previousActiveTabData.x - (tab.getBoundingClientRect().x - this.getBoundingClientRect().x);
+    } else return this.previousActiveTabData.y - (tab.getBoundingClientRect().y - this.getBoundingClientRect().y);
+  }
+
+  /**
+   * getSelectedTabScale - gets the scale of the tab
+   */
+  private getTabScale(tab: Tab): number {
+    if (this.orientation === TablistOrientation.horizontal) {
+      return this.previousActiveTabData.width / tab.getBoundingClientRect().width;
+    } else return this.previousActiveTabData.height / tab.getBoundingClientRect().height;
+  }
+
+  /**
+   * Calculates and applies updated values to CSS variables.
+   *
+   * @param tab - the tab element to apply the updated values to
+   * @internal
+   */
+  private applyUpdatedCSSValues(tab: Tab) {
+    this.calculateAnimationProperties(tab);
+    this.setTabScaleCSSVar();
+    this.setTabOffsetCSSVar();
+  }
+
+  /**
+   * Runs through all the operations required for setting the tab active indicator to its starting location, ending
+   * location, and applying the animated css class to the tab.
+   *
+   * @param tab - the tab element to apply the updated values to
+   * @internal
+   */
+  private animationLoop(tab: Tab) {
+    // remove the animated class so nothing animates yet
+    tab.setAttribute('data-animate', 'false');
+    // animation start - this applyUpdatedCSSValues sets the active indicator to the location of the previously selected tab
+    this.applyUpdatedCSSValues(tab);
+    // changing the previously active tab allows the applyUpdatedCSSValues method to calculate the correct end to the animation.
+    this.previousActiveTabData = this.activeTabData;
+    // calculate and apply updated css values for animation.
+    this.applyUpdatedCSSValues(tab);
+    // add the css class and active indicator will animate from the previous tab location to its tab location
+    tab.setAttribute('data-animate', 'true');
+  }
+
+  /**
+   * Sets the data from the active tab onto the class. used for making all the animation calculations for the active
+   * tab indicator.
+   *
+   * @internal
+   */
+  private setTabData(): void {
+    if (this.tabs && this.tabs.length > 0) {
+      const tabs = this.tabs as Tab[];
+      const activeTab = this.activetab || tabs[0];
+      const activeRect = activeTab?.getBoundingClientRect();
+      const parentRect = this.getBoundingClientRect();
+
+      this.activeTabData = {
+        x: activeRect.x - parentRect.x,
+        y: activeRect.y - parentRect.y,
+        height: activeRect.height,
+        width: activeRect.width,
+      } as TabData;
+
+      if (
+        this.previousActiveTabData?.x !== this.activeTabData?.x &&
+        this.previousActiveTabData?.y !== this.activeTabData?.y
+      ) {
+        this.previousActiveTabData = this.activeTabData;
+      }
+    }
+  }
+
+  private setTabOffsetCSSVar() {
+    this.style.setProperty('--tabIndicatorOffset', `${this.activeTabOffset}px`);
```

Comment:
SO, its preferrable to use the `$FASTController.addStyles()` instead of `style.setPorperty`.  Add Styles API will add the styles to the host element instead of to the styles property on the element itself. Also if someone set inline `styles` on this element this would collide.

```
private myStyles: ElementStyles | undefined;

private setTabOffsetCSSVar() {
   this.myStyles = css`
        :host {
          --tabIndicatorOffset: `${this.activeTabOffset}px`;
        }
      `;
      this.$fastController.addStyles(this.myStyles);
}
```

You can call removeStyles in the disconnectedCallback.

Reply:
One issue I'm seeing with generating a sheet is that it's creating a stylesheet every time a tab is clicked (see attached). To me this seems... I dunno... wasteful? Maybe memory intensive? 

<img width="1224" alt="Screenshot 2024-07-24 at 9 31 32 PM" src="https://github.com/user-attachments/assets/22da0d74-65dc-42eb-941b-bf5731f499ad">

I'm open to the change but I guess I'd like to know more about the concern that someone is going to wholesale wipeout `element.style`. When I use `setProperty` it self-repairs on the next click.

Reply:
A couple things, and this is a good convo to have IMO...

I think we want to call `.removeStyles()` before we set new styles in the method as that will avoid this continually adding of new sheets. Essentially, wipe the sheet, create the sheet.

With that said, I think that using `.addStyles()` is often preferrable but not always ideal. Sometimes it's super clear - as I'm exploring tooltip, until we get the `anchor` attribute we need to set it via CSS. It's _likely_ that the reference doesn't change often with how I'm constructing it and so I'll _probably_ use the `.addStyles()` API when `anchor` isn't available. With that said, in something like Slider, the calls to destroying and creating a new stylesheet are significant when dragging the slider and trying to update a CSS Variable value, it's happening constantly. In this case, I think the creation and removal of stylesheets is likely a recipe for trouble.

So, in this scenario, we should consider the same things - what's best for the situation, what are the possible issues with it (overriding inline styles, performance, etc) and come to what we think is best. Ultimately, this is "internals" so we can learn and iterate from it.

Reply:
I added the check above in my code sample that checks whether the styles are defined and if they are removes them before it tries to set them again. I just forgot to add that bit.

Reply:
After some perf investigation, the delta here is pretty slim on the performance angle but `setProperty` consistently pulls ahead slightly. In the grandparent method the variables get set before & after setting the active tab so we get two calls per animation loop....

| | Before Call | After Call |
|--------|--------|--------|
| addStyles | 0.1-0.3ms | 0.1-0.3ms |
| setProperty  | 0.1-0.3ms | 0ms |

At 20x CPU slowdown that upper bounds goes to ~4ms (dropped frame territory) but `setProperty` still manages to hit 0ms more often. 

The bigger story might be how much work the browser is doing...

addStyles:
![Screenshot 2024-07-31 at 1 28 14 PM](https://github.com/user-attachments/assets/939e8e87-d02e-4232-a73d-cf74b719c3d8)

setProperty:
![Screenshot 2024-07-31 at 1 29 40 PM](https://github.com/user-attachments/assets/c68afa1d-da2e-4552-88ce-1bcd4193ada2)

 You can see the flame chart for `addStyles` is a bit deeper of a callstack than for `setProperty`. Which is somewhat expected, but less work is probably better. `addStyles` also increased the surface area API of our tablist class with more imports and a `disconnectedCallback` to cleanup, which I believe one of our stated goals is to have a narrow API surface. 
 
tl;dr– No big alarm bells are going off for me with either approach, but perhaps in the interest of simplicity and marginal gains we go with `setProperty` for now. `addStyles` is essentially a cosmetic DX improvement where you don't see the `style="--foo; --bar;"` attribute on the element in DevTools but get a constructed stylesheet in the styles panel instead. 

If we decide that DX is worth it we can always reintroduce `addStyles` and we have a little bit more insight into the cost/benefits. ❤️

Reply:
Thanks for digging in here - there is a lot of really good data. I'm in favor of rolling with setProperty to kick off.